build/bin/remove_unneeded

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+# Sanity check: the file is created by the Dockerfile.
+if [ ! -f /in-the-container ]; then
+  echo "Don't run this outside the container!"
+  exit 1
+fi
+
+# Array of basic tools needed by entrypoint and user_setup.
+basicNeed=("bin/sh" "/bin/bash" "/usr/bin/coreutils")
+libraries=()
+
+# Get all that is needed by basicNeed
+for i in "${basicNeed[@]}"; do
+    ldd=$(ldd "${i}" \
+        | grep so \
```

Comment:
This pattern is broad - can we do something like `grep -P "\.so\.\d?"` so that we only match shared libraries?

Reply:
Thanks. I changed to grep "\.so\.[0-9]*"